### ISSUE TITLE
Large kmem_alloc(131072, 0x1000)

### ISSUE THREAD

**Issue by dpavlin** [zfsonlinux/zfs#3615]
When trying to create ext4 filesystem on zvol i got following in my dmesg:

```
[  217.493487] Large kmem_alloc(131072, 0x1000), please file an issue at:
               https://github.com/zfsonlinux/zfs/issues/new
[  217.493493] CPU: 3 PID: 8595 Comm: zvol Tainted: P           O    4.0.0-2-amd64 #1 Debian 4.0.8-1
[  217.493494] Hardware name: Dell Inc. Precision T3610/09M8Y8, BIOS A06 02/28/2014
[  217.493496]  0000000000000000 0000000000000000 ffffffff81562ce2 000000000000c210
[  217.493499]  ffffffffa0629cb2 ffffffffa09c7b63 ffff880076ed3858 0000000000004000
[  217.493500]  0000000000000000 ffffffffa09c7b63 000000078c000000 0000000000000000
[  217.493502] Call Trace:
[  217.493510]  [<ffffffff81562ce2>] ? dump_stack+0x40/0x50
[  217.493526]  [<ffffffffa0629cb2>] ? spl_kmem_zalloc+0x152/0x1d0 [spl]
[  217.493539]  [<ffffffffa091b079>] ? dmu_buf_hold_array_by_dnode+0x99/0x500 [zfs]
[  217.493547]  [<ffffffffa091b5bc>] ? dmu_buf_hold_array+0x5c/0x80 [zfs]
[  217.493555]  [<ffffffffa091c916>] ? dmu_write_req+0x66/0x1d0 [zfs]
[  217.493564]  [<ffffffffa09be96a>] ? zvol_write+0x10a/0x470 [zfs]
[  217.493578]  [<ffffffff81564190>] ? __schedule+0x2a0/0x920
[  217.493582]  [<ffffffffa062d2b6>] ? taskq_thread+0x1f6/0x440 [spl]
[  217.493586]  [<ffffffff81097570>] ? wake_up_state+0x20/0x20
[  217.493590]  [<ffffffffa062d0c0>] ? taskq_cancel_id+0x110/0x110 [spl]
[  217.493592]  [<ffffffff8108ab41>] ? kthread+0xc1/0xe0
[  217.493594]  [<ffffffff8108aa80>] ? kthread_create_on_node+0x180/0x180
[  217.493596]  [<ffffffff81568dd8>] ? ret_from_fork+0x58/0x90
[  217.493598]  [<ffffffff8108aa80>] ? kthread_create_on_node+0x180/0x180
[  217.494954] Large kmem_alloc(131072, 0x1000), please file an issue at:
               https://github.com/zfsonlinux/zfs/issues/new
[  217.494959] CPU: 2 PID: 8591 Comm: zvol Tainted: P           O    4.0.0-2-amd64 #1 Debian 4.0.8-1
[  217.494960] Hardware name: Dell Inc. Precision T3610/09M8Y8, BIOS A06 02/28/2014
[  217.494961]  0000000000000000 0000000000000000 ffffffff81562ce2 000000000000c210
[  217.494963]  ffffffffa0629cb2 ffffffffa09c7b63 ffff880076ed3858 0000000000004000
[  217.494965]  0000000000000000 ffffffffa09c7b63 0000000788000000 0000000000000000
[  217.494966] Call Trace:
[  217.494971]  [<ffffffff81562ce2>] ? dump_stack+0x40/0x50
[  217.494979]  [<ffffffffa0629cb2>] ? spl_kmem_zalloc+0x152/0x1d0 [spl]
[  217.494989]  [<ffffffffa091b079>] ? dmu_buf_hold_array_by_dnode+0x99/0x500 [zfs]
[  217.494997]  [<ffffffffa091b5bc>] ? dmu_buf_hold_array+0x5c/0x80 [zfs]
[  217.495005]  [<ffffffffa091c916>] ? dmu_write_req+0x66/0x1d0 [zfs]
[  217.495013]  [<ffffffffa09be96a>] ? zvol_write+0x10a/0x470 [zfs]
[  217.495015]  [<ffffffff81564190>] ? __schedule+0x2a0/0x920
[  217.495019]  [<ffffffffa062d2b6>] ? taskq_thread+0x1f6/0x440 [spl]
[  217.495022]  [<ffffffff81097570>] ? wake_up_state+0x20/0x20
[  217.495025]  [<ffffffffa062d0c0>] ? taskq_cancel_id+0x110/0x110 [spl]
[  217.495027]  [<ffffffff8108ab41>] ? kthread+0xc1/0xe0
[  217.495029]  [<ffffffff8108aa80>] ? kthread_create_on_node+0x180/0x180
[  217.495031]  [<ffffffff81568dd8>] ? ret_from_fork+0x58/0x90
[  217.495033]  [<ffffffff8108aa80>] ? kthread_create_on_node+0x180/0x180
```

This is using latest Debian packages, 0.6.4-21-53b1d9-wheezy

**Comment by behlendorf** [comment]
@dpavlin the warning is just advisory but it still needs to be explained.  Have you made any non-default tunings to the zvol?  For example, what's the record size set too?  In the meanwhile you can suppress the warning by increasing the default warning threshold which is safe.

```
echo $((256*1024)) >/sys/module/spl/parameters/spl_kmem_alloc_warn
```

**Comment by dpavlin** [comment]
I'm creating filesystem with

```zfs create -V 60G -b 4k -s klinz/block/blabla```

However, I'm not getting warning in dmesg after that command, but instead after

```mkfs.ext4 -m 0 -L blabla /dev/zvol/klinz/block/blabla```

I also noticed that mkfs.ext4 does finish and create file-system, but label never appears in ```/dev/disk/by-label/```. I'm not sure if this is zfs related or not, but it did work correctly with last zfs Debian package revision. Kernel used is ```Linux klin 4.0.0-2-amd64 #1 SMP Debian 4.0.8-1 (2015-07-11) x86_64 GNU/Linux``` from Debian sid. I don't think I updated it since it worked list time, and even if I did it was some version, but newer build.

I have script which waits for label of new file-system to appear, so this is somewhat a show-stopper for me.

With sparse and default volblocksize of 8K i'm getting also warning for for smaller allocation: ```Large kmem_alloc(65536, 0x1000)``` and same backtrace.

Removing sparse option to zfs create doesn't change warnings.

**Comment by dpavlin** [comment]
I tried similar test on ```Linux mlin 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1 (2015-05-24) x86_64 GNU/Linux``` on Debian jessie (current stable) and I'm not seeing any warnings in dmesg nor problems with labels. So it seems that it's a kernel problem (perhaps) and not zfsonlinux one. Feel free to close this issue if you agree with this.

**Comment by prometheanfire** [comment]
should this be combined with #3684

**Comment by behlendorf** [comment]
@prometheanfire yes, yes it should. Thanks for noticing this one, this is a duplicate of #3684 and was resolved by commit c495fe2.
